src/Ast.ml

```diff
@@ -500,11 +500,51 @@ end
 
 include T
 
+let attributes = function
+  | Pld _ -> assert false
+  | Typ x -> x.ptyp_attributes
+  | Cty x -> x.pcty_attributes
+  | Pat x -> x.ppat_attributes
+  | Exp x -> x.pexp_attributes
+  | Cl x -> x.pcl_attributes
+  | Mty x -> x.pmty_attributes
+  | Mod x -> x.pmod_attributes
+  | Sig _ -> assert false
+  | Str _ -> assert false
+  | Top -> assert false
```

Comment:
I'm not sure that it's better to crash in these cases instead of returning the empty list. AFAIU e.g. signatures do not currently have attributes in the parsetree, but it would make sense if they did.